src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -193,4 +204,88 @@ object BoringUtils {
     sinks.foreach(addSink(_, genName, true, true))
     genName
   }
+
+  /** Access a source [[Data]] that may or may not be in the current module.  If
+    * this is in a child module, then create ports to allow access the
+    * requrested source.
+    */
+  def bore(source: Data)(implicit si: SourceInfo): Data = {
+    import reflect.DataMirror
+    def parent(d: Data): experimental.BaseModule = d.topBinding.location.get
+    def purePortType = if (DataMirror.hasOuterFlip(source)) Flipped(chiselTypeOf(source)) else chiselTypeOf(source)
+    def emptyWireOnError = Wire(purePortType) //return empty wire to continue elaboration to collect other errors
```

Comment:
Can this return `DontCare` or is that problematic due to that not working with zero-width?

Reply:
Hmm. This is only returned on the case when we are definitely throwing an error, but probably `DontCare` will work, I'll try it.